lib/space.py

```diff
@@ -727,6 +745,19 @@ def region_factory(space=None, NW=None, NE=None, SW=None,
                    **kwargs):
     region_name = gen_region_name(NW=NW, NE=NE, SW=SW, SE=SE,
                                   center=center, size=size)
+    if 'model_name' in kwargs and 'exec_key' in kwargs:
+        model_name = kwargs['model_name']
+        exec_key = kwargs['exec_key']
+        if exec_key is not None and model_name is not None:
+            region_name = gen_region_name_for_model_at_exec_key(model_name,
+                                                                exec_key,
+                                                                NW=NW,
+                                                                NE=NE,
+                                                                SW=SW,
+                                                                SE=SE,
+                                                                center=center,
+                                                                size=size)
+
```

Comment:
@gcallah we could probably rethink having a global region dictionary. 
The model which sets up the region in this global registry first will always be there and we wont create a region for another model.
Region overlaps between models sometimes. Region setup by model A can conflict with model B so that when model B retrieves it's neighbors it get's neighbors for model A.